### DISCUSSION TITLE
SSerialSend with esp32

### DISCUSSION THREAD

**Discussion by VBill45** [arendst/Tasmota#21313] [Q&A]
Hello

I would like to replace an esp01 (8266) with an esp32s2 mini on an LC Technology 2 relay board.
To make the best use of the connection I use the 3.3v GPIO 14 configured as RX and GPIO 9 configured as TX of the esp32s2 mini + a wire to connect the GND.

This way the ESP32s2 mini mounts directly onto the connector on the relay board that was used for the esp01.

I set tasmota parameters:
Set GPIO2 and GPIO3 as Relay1 and Relay 2 (in module configuration and click save)
Disable SerialLog "seriallog 0".
Add the following rules : Rule1 on Power1#State=1 do SerialSend5 A00101A2 endon on Power1#State=0 do SerialSend5 A00100A1 endon on Power2#State=1 do SerialSend5 A00201A3 endon on Power2#State=0 do SerialSend5 A00200A2 endon

Enable the rule "rule1 1"

But it doesn't work

Now I think I need to use the soft serial link, but the SSerialSend command is returned as unknown

Does anyone know why or have a solution?

PS of course I've already tried changing the baud rate and it's checked on my esp01 which is at 115200 and works fine.

PS 2 the change to an esp32s2 mini is to connect more contacts which is not possible with the esp01

Bonjour

Je voudrais remplacer un esp01 (8266) par un esp32s2 mini sur une platine 2 relai LC Technology.
Pour utiliser au mieux la connection j'utilise le 3.3v le GPIO 14 configuré en RX et le GPIO 9 configuré en TX du esp32s2 mini + un fil pour connecter le GND.

Comme ça le ESP32s2 mini se monte directement sur le connecteur de la platine relai qui servait pour le esp01.

J'ai paramétré tasmota :
Set GPIO2 and GPIO3 as Relay1 and Relay 2 (in module configuration and click save)
Disable SerialLog "seriallog 0"
Add the following rules : Rule1 on Power1#State=1 do SerialSend5 A00101A2 endon on Power1#State=0 do SerialSend5 A00100A1 endon on Power2#State=1 do SerialSend5 A00201A3 endon on Power2#State=0 do SerialSend5 A00200A2 endon

Enable the rule "rule1 1"

Mais ça ne fonctionne pas

La, je me dis qu'il faut utiliser la liaison serie soft mais la commande SSerialSend est renvoyé comme inconnu

Si quelqu'un sait pourquoi ou a une solution ?

PS biensur j'ai déja essayé de changer la vitesse de transmission et est vérifié sur mon esp01 qui est à 115200 et fonctionne très bien

PS 2 le passage a un esp32s2 mini est pour connecter des contacts en plus ce qui n'est pas possible avec le esp01

**Comment by Jason2866** [comment]
Esp32 does not use software serial. All serial are using hardware serial. So the command sserial is not available on any esp32
